[W5.11][F09-A2] Thanh Nguyen

kokonguyen191 opened the pull request:
Make FindCommand case-insensitive
Update I/O and JUnit tests to match case-insensitive Find, update UserGuide

kokonguyen191 committed:
Make FindCommand case-insensitive
Update I/O and JUnit tests to match case-insensitive Find, update UserGuide

jonathanwj commented on src/seedu/addressbook/commands/FindCommand.java:
```diff
@@ -5,19 +5,21 @@
 import java.util.HashSet;
 import java.util.List;
 import java.util.Set;
+import java.util.stream.Collectors;
```
It is better to remove unused import statements.

kokonguyen191 committed:
Remove unused imports

jonathanwj commented on src/seedu/addressbook/common/Utils.java:
```diff
@@ -34,4 +37,13 @@ public static boolean elementsAreUnique(Collection<?> items) {
+    public static boolean isCaseInsensitiveMatched(Set<String> firstSet, Set<String> secondSet) {
+        List<String> firstSetInLowerCase = firstSet.stream().map(String::toLowerCase).collect(Collectors.toList());
+        List<String> secondSetInLowerCase = secondSet.stream().map(String::toLowerCase).collect(Collectors.toList());
+        return Collections.disjoint(firstSetInLowerCase, secondSetInLowerCase);
```
Shouldn't this return true if set of strings are disjoint? 
Collections.disjoint(..) returns true if the two specified collections have no elements in common.

Header comments does not match.

kokonguyen191 replied:
Oh yea I didn't notice that!

kokonguyen191 committed:
Fix isCaseInsensitiveMatched method

kokonguyen191 committed:
Fix getPersonsWithNameContainingAnyKeyword method

Line 54: From adding a result if wordsInName and keywords don't match to adding a result if wordsInName and keywords match.

kokonguyen191 committed:
Remove unused imports
Reformat to Google code style

k-l-a commented on src/seedu/addressbook/common/Utils.java:
```diff
@@ -34,4 +38,15 @@ public static boolean elementsAreUnique(Collection<?> items) {
     }
+
+    /**
+     * Return true if two set of strings are not disjoint, regardless of capitalization.
```
Should be `Returns`.

k-l-a replied:
Should be `Returns`

k-l-a commented on src/seedu/addressbook/common/Utils.java:
```diff
@@ -34,4 +38,15 @@ public static boolean elementsAreUnique(Collection<?> items) {
+    /**
+     * Return true if two set of strings are not disjoint, regardless of capitalization.
+     */
+    public static boolean isCaseInsensitiveMatched(Set<String> firstSet, Set<String> secondSet) {
```
Good job on extracting the method. Maybe u can extract it even more by having a method that compares the strings themselves (instead of just the sets), and use that within this method.

k-l-a replied:
Good job on extracting the method. Maybe u can extract it even more by having a method that compares the strings themselves (instead of just the sets), and use that within this method.

k-l-a reviewed (commented):
@kokonguyen191 Some comments added. Please close PR after reading.

Also, you should make commits in small chunks and not in one big chunk like your initial commit. This would defeat the purpose of a RCS.
